assets/src/edit-story/components/library/panes/media/common/mediaElement.js

```diff
@@ -169,9 +170,13 @@ const MediaElement = ({
     mediaElement?.current?.getBoundingClientRect();
 
   const dropTargetsBindings = useMemo(
-    () => ({
+    () => (thumbnailURL) => ({
       draggable: 'true',
       onDragStart: (e) => {
+        resourceList[resource.id] = {
```

Comment:
This resourceList is effectively a global variable, isn't it? :/ Couldn't the cache be instead stored in a top-level redux store? A ResourceCache component in the redux tree?

Reply:
> This resourceList is effectively a global variable, isn't it? :/ Couldn't the cache be instead stored in a top-level redux store? A ResourceCache component in the redux tree?

It could, but what would be the benefit? I just see a performance hit. We already re-rendering too much on every editor interaction, with this PR solution, only the image/display.js part is re-rendered once we have the full-size image.